### ISSUE TITLE
server running but players get kicked

### ISSUE THREAD

**Issue by ghost** [ravetroll/BedrockService#3]
The installation seemed to go easy enough but after logging into the game server as a local player I get kicked within 15-20 seconds. Did I miss a step?

--edit-- Same for WAN clients

**Comment by ravetroll** [comment]
Because backups are on by default, it may be that the system is taking an initial backup.  Try turning backups off in the config file to see if that fixes it.  When you run bedrock_server.exe directly do people get kicked or does it work fine?

**Comment by ghost** [comment]
It works fine

**Comment by ghost** [comment]
how long should the initial back up take?

**Comment by ravetroll** [comment]
The backup takes as long as it takes to copy the worlds folder to another folder.

Have you tried running with the backups turned off?

**Comment by ghost** [comment]
I haven’t yet. I am in the process of testing now with one.

**Comment by ravetroll** [comment]
If you make sure the service is stopped and then run bedrockservice.exe, you will be able to see what its doing in a command window.  To test backups, set the intervalMinutes to 1 and watch what it says in the command window.  You should be able to see bedrock_server.exe messages indicating its starting and stopping every couple of minutes.  If that appears to be working, then stop it, reset the intervalMinutes to something more sensible like 1440 for a day or 10080 for a week, and check to make sure it doesn't keep trying to stop and start every couple of minutes.  If it tries to stop and start every couple of minutes even though the interval is a lot longer there may be in an incompatibility with the filesystem not reporting the modified time correctly on the backup directory.

**Comment by ghost** [comment]
OK. I will do that. I will know something late this AM.

**Comment by ghost** [comment]
OK, I got a chance to test this and it works perfect after changing the config to say Backup=false.  The timer was already set to 1440. So are you saying that when I initially installed it with the timer set at 1440 in the config it is normal for players to get kicked until  the backup is complete?  

So I can just run with backup false for the time being. That’s not a problem. 

How about multiple servers at once. What all lines do I need to change in the config and to keep up with it easier can I rename the .exe to match the server name?

**Comment by ravetroll** [comment]
The way the backup function works is that it checks to see if a backup exists within the interval period from the current time.  The interval period is specified in minutes (1440 implys a daily backup).  When you start up for the first time and have no backups, then it will stop the server and create a backup and then restart.  So when you run it for the first time i would expect it to kick players connecting immediately.  once it has got that initial backup out of the way it should not kick players until the next backup is due.

I have not designed it to be able to run multiple servers and I don't think it would be possible due to a naming conflict if you tried to run multiple instances.  Something to look at adding I guess.

**Comment by ravetroll** [comment]
I have redesigned the app to support running multiple bedrock_server.exe instances.  There are some configuration changes required as I have restructured the configuration section.  Backups also work differently now, but it is possible for backups to work for all running instances too.  Only 1 instance of BedrockService is required to run multiple instances of bedrock_server.exe.  Let me know how it goes for you.
